### ISSUE TITLE
FakeAIOKafkaConsumer.getone` returns `None` but should block

### ISSUE THREAD

**Issue by PeterJCLaw** [alm0ra/mockafka-py#117]
The real `aiokafka` consumer's `getone` implementation blocks indefinitely rather than returning `None` when there are no messages. Currently `FakeAIOKafkaConsumer.getone` differs from that.

Spotted while working on #112.

**Comment by alm0ra** [comment]
Dear @PeterJCLaw,

Based on [this implementation](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py) and the comment in aiokafka:
"If no new messages prefetched, this method will wait for it."

It's easy to handle this approach and make getone blocking. However, in Mockafka, there is no way to inject messages from anywhere. If we block it, it will wait forever. The Kafka layer and the consumer layer are tightly coupled, and there isn't a separate section for handling Kafka.

If we could decouple Mockafka to function like a Kafka socket and use Mockafka as a lightweight Kafka, we could achieve this. However, in this implementation, I think it's acceptable as is.

**Comment by PeterJCLaw** [comment]
> Dear @PeterJCLaw,
> 
> Based on [this implementation](https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/consumer.py) and the comment in aiokafka: "If no new messages prefetched, this method will wait for it."

Indeed. I actually think it's unfortunate that aiokafka doesn't offer a timeout on that method.

> It's easy to handle this approach and make getone blocking. However, in Mockafka, there is no way to inject messages from anywhere. If we block it, it will wait forever. The Kafka layer and the consumer layer are tightly coupled, and there isn't a separate section for handling Kafka.

Yeah, there isn't an obviously correct fix here. A couple of other approaches we could consider:
- raise a custom exception, which in its message explains why it exists and that it's a testing artefact rather than a real simulation of the API
- let this be configurable, perhaps letting callers pass a function to `FakeAIOKafkaConsumer.__init__` which describes the desired behaviour (perhaps defaulting to raising the above exception); regardless of what we do this could be added later if needed

I think either of these is probably going to be clearer to the user than an `AttributeError` from the returned `None` which might mislead them about the behaviour of aiokafka's API. (Yes, I hope they'd read the aiokafka docs too, but the reality is that people often don't)

I'd also be tempted to look at what e.g: `fakeredis` does for its faking of any similar APIs; Redis has pub/sub, so there might be some prior art we could consider.

**Comment by alm0ra** [comment]
> * raise a custom exception, which in its message explains why it exists and that it's a testing artefact rather than a real simulation of the API
I believe that raising custom exceptions is not a behavior that aiokafka has, and it does not provide a good experience when working with this library.

> * let this be configurable, perhaps letting callers pass a function to `FakeAIOKafkaConsumer.__init__` which describes the desired behaviour (perhaps defaulting to raising the above exception); regardless of what we do this could be added later if needed

I agree with this. Making it configurable is a good approach. We could default to raising the exception, but how should we handle this in a way that aligns with aiokafka behavior? Should we wait indefinitely for a message or print a warning? Let's discuss the best approach.

**Comment by PeterJCLaw** [comment]
I agree that raising an exception isn't what `aiokafka` does, however if we're agreed that we want to deviate from the library's blocking behaviour then the question is _how_ to deviate rather than whether to or not.

I therefore see the choices as being:
- return some dummy value which is eventually likely to raise an error in the user's code, most probably an `AttributeError`, leaving the user to investigate for themselves what happened
- raise an explicit error which contains a message we control about why the behaviour is different, how to fix it, etc.

I see the latter as providing the better experience. While either way the user gets an error they might not want, in the latter case we can ensure that the reasons for the error and how to resolve it are clear.

Would be great to understand if/how you see this differently.
